#170 AdnRelax page 1.5 (#173)
Inbibo/adonisfx_docs: docs/maya/deformers/relax.md

# AdnRelax

> AdnRelax is a Maya deformer designed to smooth creases and correct over-compression or over-stretching on geometry surfaces. This deformer can help refining the fascia and skin meshes resulting from the simulation by applying a combination of smoothing, relaxation, and volume correction algorithms. The AdnRelax deformer applies smoothing and relaxation in each iteration, while push in and push out adjustments are applied during the final iteration.

Reviewer: ```suggestion
AdnRelax is a Maya deformer designed to smooth creases and correct over-compression or over-stretching on geometry surfaces. This deformer can help refining the fascia and skin meshes resulting from the simulation by computing an iterative algorithm that combines smoothing, relaxation, and volume corrections. The AdnRelax deformer applies smoothing and relaxation in each iteration, while the volume correction (i.e. a push in and push out adjustment in the direction of the point normals) is applied during the final iteration.
```

Author: Done.

> AdnRelax is a Maya deformer designed to smooth creases and correct over-compression or over-stretching on geometry surfaces. This deformer can help refining the fascia and skin meshes resulting from the simulation by applying a combination of smoothing, relaxation, and volume correction algorithms. The AdnRelax deformer applies smoothing and relaxation in each iteration, while push in and push out adjustments are applied during the final iteration.

Author: Right, done.

### How to use

> The AdnRelax deformer is easy to create and configure in Maya. Ann AdnRelax deformer requires the following input:

Reviewer: ```suggestion
The AdnRelax deformer is easy to create and configure in Maya. An AdnRelax deformer requires the following input:
```

Author: Fixed.

> The AdnRelax deformer is easy to create and configure in Maya. Ann AdnRelax deformer requires the following input:

Author: Done.

> **Input Mesh (I)**: The mesh to apply the relaxation algorithm onto. This should typically be a fascia or skin mesh already simulated.

Reviewer: I feel that we have to stop using this kind of achronyms for when the inputs are not complicated. This helped especially for AdnSimshape. I think here we can just say:

"The AdnRelax deformer is easy to create and configure in Maya. It only requires the mesh to apply the realaxation onto. Typically, this mesh would be the simulated fascia or skin."

Author: Agreed!

> **Input Mesh (I)**: The mesh to apply the relaxation algorithm onto. This should typically be a fascia or skin mesh already simulated.

Author: Done.

To create an AdnRelax deformer, follow these steps:

1. Select the **Input Mesh**.

> 2. Press ![Relax button](../images/adn_relax.png){style="width:4%"} in the AdonisFX shelf or *Relax* in the AdonisFX menu, under the Create Solvers section.

Reviewer: This reminds me that we probably have to fix the AdnSkinMerge. Could you please double check?

```suggestion
2. Press ![Relax button](../images/adn_relax.png){style="width:4%"} in the AdonisFX shelf or *Relax* in the AdonisFX menu, under the Create Deformers section.
```

Author: Fixed. This was already correct for AdnSkinMerge.

> 3. A message in the terminal will notify you that AdnRelax has been created properly, meaning that it is ready to deform the mesh with default settings. Check the next section to customize their configuration.

Reviewer: Please add a note here. It is correct what you say "ready to deform the mesh with default settings" Probelm is that defaults have iterations set to 0, so user won't see much. I would consider to reformulate saying something like "the deformer is ready to use, increase the number of iterations to see the effect of the deformation"

Author: You are right. I have reformulated it.

> 3. A message in the terminal will notify you that AdnRelax has been created properly, meaning that it is ready to deform the mesh with default settings. Check the next section to customize their configuration.

Reviewer 2: I would also add a reference to the section instead of saying next section.
```suggestion
3. A message in the terminal will notify that AdnRelax has been created properly, meaning that it is ready to deform the mesh with default settings. Check the next section to customize their configuration.
```

Author: Done.

## Attributes

### Settings
| Name | Type | Default | Animatable | Description |
| :--- | :--- | :------ | :--------- | :---------- |
| **Iterations**         | Integer | 0      | X | Number of iterations of the relaxation algorithm. Greater values mean greater computational cost. Has a range of \[1, 10\]. The upper limit is soft, higher values can be used. |
| **Pin**                | Boolean | False  | ✓ | Flag to pin the vertices on the boundaries. |
| **Smooth**             | Float   | 0.5    | ✓ | Amount of smoothing to apply. Has a range of \[0.0, 1.0\]. |
| **Relax**              | Float   | 0.5    | ✓ | Amount of relaxation to apply. Has a range of \[0.0, 1.0\]. |
| **Push In Ratio**      | Float   | 0.0    | ✓ | Amount of correction applied by the push in adjustment. Has a range of \[0.0, 2.0\]. The upper limit is soft, higher values can be used. |
| **Push In Threshold**  | Float   | -1.0   | ✓ | Maximum correction applied by the push in adjustment. The threshold will be ignored if its value is 0.0 or less. Has a range of \[-1.0, 2.0\]. The upper limit is soft, higher values can be used. |

> | **Push Out Ratio**     | Float   | 0.0    | ✓ | Amount of correction applied by the push out adjustment. Has a range of \[0.0, 2.0\]. The upper limit is soft, higher values can be used. |

Reviewer 2: If the upper limits are soft should we use `[` instead of `]`? Eg. \[-1.0, 2.0\[

Author: Well the thing is that in all pages we always use `[x, y]` regardless of whether the upper limit is soft or not. So that change would have to be applied to all the pages.
